documentation/modules/oauth/con-oauth-authentication-broker.adoc

```diff
@@ -0,0 +1,138 @@
+// Module included in the following assemblies:
+//
+// assembly-oauth.adoc
+
+[id='con-oauth-authentication-broker-{context}']
+= {oauth} Kafka broker configuration
+
+Kafka broker configuration for {oauth} involves:
+
+* Creating the {oauth} client in the authorization server
+* Configuring {oauth} authentication in the Kafka custom resource
+
+NOTE: In relation to the authorization server, Kafka brokers and Kafka clients are both regarded as {oauth} clients.
+
+== {oauth} client configuration on an authorization server
+
+To configure a Kafka broker to validate the token received during session initiation,
+the recommended approach is to create an {oauth} _client_ definition in an authorization server, configured as _confidential_, with client credentials enabled:
+
+* Client ID of `Kafka`
+* Client ID and Secret as the authentication mechanism
+
+NOTE: You only need to use a client ID and secret when using a non-public introspection endpoint of the authorization server.
+The credentials are not typically required when using public authorization server endpoints, as with fast local JWT token validation.
+
+== {oauth} authentication configuration in the Kafka cluster
+
+To use {oauth} authentication in the Kafka cluster, you specify a TLS listener configuration for your Kafka cluster custom resource with the authentication method `oauth`:
+
+.Authentication method type for {oauth}
+[source,yaml,subs="+quotes, attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+spec:
+  kafka:
+    listeners:
+      *tls*:
+        authentication:
+          type: *oauth*
+          #...
+----
+
+You can configure `plain`, `tls` and `external` listeners, as described in xref:assembly-configuring-kafka-broker-listeners-deployment-configuration-kafka[Kafka broker listeners],
+but it is recommended not to use `plain` listeners with {oauth} as this creates a vulnerability to network eavesdropping and unauthorized access through token theft.
+
+You configure an `external` listener with `tls: true` for a secure transport layer to communicate with the client.
+
+.External listener with TLS
+[source,yaml,subs="+quotes"]
+----
+# ...
+listeners:
+  tls:
+    authentication:
+      type: oauth
+  external:
+    type: oauth
+    *tls: true*
+    authentication:
+      type: tls
+# ...      #...
```

Comment:
This should be something like this:

```yaml
# ...
listeners:
  tls:
    authentication:
      type: oauth
  external:
    type: loadbalancer
    authentication:
      *type: oauth*
# ...
```

(Sorry, GitHub still cannot do multiline suggestions)

Reply:
Workaround: If you make a multiline suggestion, then cancel, the lines are still selected in the source. Then you can choose '+' and make your multiline suggestion.

Reply:
We introduce this section, mentioning `tls:true` -- not needed?

Reply:
It is `true` by default. So it is not needed. If you keep it, maybe we at least don't need it to be bold.

Reply:
Okay. I've kept it and mentioned that it's true by default, so it can be left off.